tests/operators/test_agnostic_stats_check.py

```diff
@@ -28,154 +29,94 @@
-        tables_objects = create_tables_objects(tables)
-        try:
-            a = aql.stats_check(
-                main_table=tables_objects[0],
-                compare_table=tables_objects[2],
+@pytest.mark.parametrize(
+    "sql_server",
+    [
+        "snowflake",
+        "postgres",
+        "bigquery",
+    ],
+    indirect=True,
+)
+@pytest.mark.parametrize(
+    "test_table",
+    [
+        [
+            {
+                "path": str(CWD) + "/../data/homes.csv",
+                "load_table": True,
+                "is_temp": False,
+                "param": {
+                    "schema": SCHEMA,
+                    "table_name": test_utils.get_table_name("test_stats_check_1"),
+                },
+            },
+            {
+                "path": str(CWD) + "/../data/homes3.csv",
+                "load_table": True,
+                "is_temp": False,
+                "param": {
+                    "schema": SCHEMA,
+                    "table_name": test_utils.get_table_name("test_stats_check_2"),
+                },
+            },
+        ],
+    ],
+    indirect=True,
+)
+def test_stats_check_postgres_outlier_exists(sample_dag, sql_server, test_table):
+    with sample_dag:
+        aql.stats_check(
+            main_table=test_table[0],
+            compare_table=test_table[1],
+            checks=[aql.OutlierCheck("room_check", {"rooms": "rooms"}, 2, 0.0)],
+            max_rows_returned=10,
+        )
+        test_utils.run_dag(sample_dag)
+
+
+@pytest.mark.parametrize(
+    "sql_server",
+    [
+        "snowflake",
+        "postgres",
+        "bigquery",
+    ],
+    indirect=True,
+)
+@pytest.mark.parametrize(
+    "test_table",
+    [
+        [
+            {
+                "path": str(CWD) + "/../data/homes.csv",
+                "load_table": True,
+                "is_temp": False,
+                "param": {
+                    "schema": SCHEMA,
+                    "table_name": test_utils.get_table_name("test_stats_check_1"),
+                },
+            },
+            {
+                "path": str(CWD) + "/../data/homes2.csv",
+                "load_table": True,
+                "is_temp": False,
+                "param": {
+                    "schema": SCHEMA,
+                    "table_name": test_utils.get_table_name("test_stats_check_2"),
+                },
+            },
+        ],
+    ],
+    indirect=True,
+)
+def test_stats_check_outlier_exists(sample_dag, sql_server, test_table):
+    with pytest.raises(BackfillUnfinished):
+        with sample_dag:
+            aql.stats_check(
+                main_table=test_table[0],
+                compare_table=test_table[1],
                 checks=[aql.OutlierCheck("room_check", {"rooms": "rooms"}, 2, 0.0)],
                 max_rows_returned=10,
             )
-            a.execute({"run_id": "foo"})
-            assert True
-        except ValueError as e:
-            assert False
+        test_utils.run_dag(sample_dag)
```

Comment:
Is there any other type of meaningful assert we could add to this test?

Reply:
Yes, we should I'll update.